### ISSUE TITLE
parentheses inside string in conditional clause not recognized as such

### ISSUE THREAD

**Issue by Konfekt** [koalaman/shellcheck#2226]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1029
- My shellcheck version (`shellcheck --version` or "online"): 0.7.2
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

https://github.com/cykerway/complete-alias/raw/master/complete_alias

#### Here's what shellcheck currently says:

```
Line 98:
            elif [[ '(' == "${str:j:1}" ]]; then
            ^-- SC1009: The mentioned syntax error was in this elif clause.
                 ^-- SC1073: Couldn't parse this test expression. Fix to allow more checks.
                    ^-- SC1029: In [[..]] you shouldn't escape ( or ).
                        ^-- SC1108: You need a space before and after the == .
                            ^-- SC1072: Expected comparison operator (don't wrap commands in []/[[]]). Fix any mentioned problems and try again.
```

#### Here's what I wanted or expected to see:

```
```

**Comment by tyleralbee** [comment]
I'm seeing SC1029 for the same reason. 

See [line 337 in complete-alias](https://github.com/cykerway/complete-alias/blob/7f2555c2fe7a1f248ed2d4301e46c8eebcbbc4e2/complete_alias#L337):

```bash
elif [[ '(' == "${str:j:1}" ]]; then
```

Shellcheck doesn't return SC1029 if the conditional order is swapped:

```bash
elif [[ "${str:j:1}" == '(' ]]; then
```
